### PR TITLE
Python venv setup guide

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
This is intended to be comprehensible by someone with no prior experience of Python package management.

### CODE REVIEW

File: kb/development/python-environment.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+Initial setup:
+
+```sh
+python3 -m venv ~/venv-mbedtls-development
+~/venv-mbedtls-development/bin/python scripts/min_requirements.py
+```
+
+Day-to-day work: start a shell and issue this command:
+```sh
+. ~/venv-mbedtls-development/bin/activate
```
Comment (gilles-peskine-arm) [inline kb/development/python-environment.md:24]: Should you unset `PYTHONPATH` as well? This can cause disruption sometimes.

File: kb/development/python-environment.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+## Set up a Python package environment
+
+### Quick setup
+
+Initial setup:
+
+```sh
+python3 -m venv ~/venv-mbedtls-development
+~/venv-mbedtls-development/bin/python scripts/min_requirements.py
```
Comment (yanesca) [inline kb/development/python-environment.md:19]: Is this different from
```sh
python3 -m venv ~/venv-mbedtls-development
. ~/venv-mbedtls-development/bin/activate
python scripts/min_requirements.py
```

Comment (gilles-peskine-arm) [reply kb/development/python-environment.md:19]: It's the same. Running `activate` in a shell can't really be undone, so I don't want to show instructions to run it without starting a separate shell.

Comment (yanesca) [reply kb/development/python-environment.md:19]: Shouldn't `deactivate` undo it?

File: kb/development/python-environment.md (hunk 1 above)
Comment (yanesca) [reply kb/development/python-environment.md:24]: I don't think that should be in the basic instructions, but might worth mentioning in a remark.

**Review by yanesca — APPROVED** [review]
Left some questions/comments, but looks good to me already as it is.

File: kb/development/python-environment.md (hunk 1 above)
Comment (minosgalanakis) [reply kb/development/python-environment.md:24]: With regards to PYTHONPATH I believe nowadays venv is unsetting `PYTHONHOME`. 

But there is nothing stopping users from adding an extra unset for it in ~/venv-mbedtls-development/bin/activate

It falls outside of the scope of this document though. If needed we could have a remark in line of 

> If required `$PYTHONPATH` and `$PYTHONHOME` can be configured or unset by modifying `~/venv-mbedtls-development/bin/activate`

File: kb/development/python-environment.md (hunk 2 above)
Comment (gilles-peskine-arm) [reply kb/development/python-environment.md:19]: Ah, it should. I'm not in the habit of using it, I don't remember if it's because it used to not exist or it used to not work well or I just don't like the risk when starting a new shell is so easy.

File: kb/development/python-environment.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+```sh
+. ~/venv-mbedtls-development/bin/activate
+```
+The same `activate` script works in zsh and ksh as well. There are also `activate.*` scripts for fish, csh and powershell.
+
+Now you can run e.g. `tests/scripts/check-python-files.sh` with the same tool versions as the CI.
+
+### Installing different Python versions
+
+Each virtual environment can have a different Python version, and running in a virtual environment makes that version of Python the one on the command search path. But all the Python versions need to be installed at the system level.
```
[the remark this answers is not in the document]
Comment (gilles-peskine-arm) [reply kb/development/python-environment.md:54]: The previous section already discusses setting up separate virtual environments with different python versions, so I didn't think it was worth going in detail here. But it is a little hard to follow in the previous section as well, so I'll massage the text a bit.

[the remark this answers is not in the document]
Comment (gilles-peskine-arm) [reply kb/development/python-environment.md:54]: `venv` is part of the standard library. You don't need a separate package. The Debian `python3.N-venv` package just has a wrapper script that runs `venv.main()`.

**Review by minosgalanakis — APPROVED** [review]
LGTM

**Review by yanesca — APPROVED** [review]
LGTM
